src/main/java/net/neoforged/neoforge/resource/VanillaServerListeners.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.resource;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.ServerAdvancementManager;
+import net.minecraft.server.ServerFunctionLibrary;
+import net.minecraft.server.packs.resources.PreparableReloadListener;
+import net.minecraft.world.item.crafting.RecipeManager;
+import net.neoforged.neoforge.common.util.VanillaClassToKey;
+import net.neoforged.neoforge.event.AddReloadListenerEvent;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Keys for vanilla {@link PreparableReloadListener reload listeners}, used to specify dependency ordering in the {@link AddReloadListenerEvent}.
+ * <p>
+ * Due to the volume of vanilla listeners, these keys are automatically generated based on the class name.
+ * 
+ * @see {@link VanillaClientListeners} for vanilla client listener names.
+ * @see {@link NeoForgeReloadListeners} for Neo-added listener names.
+ */
+public class VanillaServerListeners {
```

Comment:
`final` + private ctor

Reply:
Honestly both of those things are pointless. Extensions of this class are meaningless, and so would be instantiations.

I'll stand by not bothering with either unless there is a necessity for them.

Reply:
It's just good practice, and if we ever get around to generating javadoc it will at least stop generating the constructor.

Reply:
I think it's noise - whenever I encounter those constructs I stop to think "why is this class final", because it was a _purposeful_ choice to make it final, even though it's irrelevant in context. 

These constructs should be reserved for when they have meaning, either because some class hierarchy requires that the class not be extended, or because a class can only be instantiated locally for logical constraints.

We can open up a broader discussion later, at which point we can evaluate if this is a stylistic choice we would like to apply across the board, and if we can automate it. For now, I will elect to close this discussion and not make this change.